File: Sorting/Insertion Sort/Ruby/Insertion_Sort.rb
def insertion_sort(array)
    for i in 1..(array.length-1)
        key = array[i]
        j = i-1
        while j >= 0 and key < array[j]
            array[j+1] = array[j]
            j = j - 1
        end
        array[j+1] = key
    end
end


# Dummy data for testing 
test_array = [7, 6, 5, 4, 3, 2, 1]
# Display unsorted array
puts "Unsorted Array:"
puts test_array.join(' ')
# Perform insertion sorting
insertion_sort(test_array)
# Display the sorted results 
puts "Post Insertion Sort"
puts test_array.join(' ')

File: Sorting/Insertion Sort/JavaScript/Insertion_Sort.js
function insertionSort (array) {
    for (var i = 0; i < array.length; i++) {
        var temp = array[i];
        for (var j = i - 1; j >= 0 && (array[j] > temp); j--) {
            array[j + 1] = array[j];
        }
        array[j + 1] = temp;
    }
}
